server/sonar-server/src/test/java/org/sonar/server/rule/index/RuleIndexMediumTest.java

```diff
@@ -331,13 +323,12 @@ public void search_by_has_debt_characteristic() throws InterruptedException {
       .setDefaultRemediationFunction(null));
     dbSession.commit();
 
-    // 0. assert base case
     assertThat(index.search(new RuleQuery(), new QueryContext()).getTotal()).isEqualTo(2);
-    assertThat(db.debtCharacteristicDao().selectCharacteristics()).hasSize(2);
 
-    // 1. assert hasSubChar filter
     assertThat(index.search(new RuleQuery().setHasDebtCharacteristic(true), new QueryContext()).getTotal())
       .isEqualTo(2);
+    assertThat(index.search(new RuleQuery().setHasDebtCharacteristic(false), new QueryContext()).getTotal())
```

Comment:
I would add a rule with no characteristic in the fixture to check that the filter works as expected.